.github/workflows/pr.yaml

```diff
@@ -9,7 +9,54 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
+env:
+  OTEL_SERVICE_NAME: 'pr-rmm'
+  # TODO: this should be set as an org-wide variable
+  OTEL_EXPORTER_OTLP_ENDPOINT: https://tempo.gha-runners.nvidia.com:4318
+  OTEL_EXPORTER_OTLP_PROTOCOL: "http/protobuf"
+  OTEL_RESOURCE_ATTRIBUTES: "git.repository=${{github.repository}},git.ref=${{github.ref}},git.sha=${{github.sha}},git.job_url=${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+  # Set these to point the shared-actions clones at a branch on the rapidsai/shared-actions repo
+  SHARED_ACTIONS_REPO: "rapidsai/shared-actions"
+  SHARED_ACTIONS_REF: "main"
+
 jobs:
+  reexports:
+    # Re-export secrets and env vars to make them available to shared actions and workflows
+    # You can't use env.SOMETHING in a "with:" section when calling a shared workflow.
+    runs-on: ubuntu-latest
+    outputs:
+        endpoint: "${{steps.reexport.outputs.endpoint}}"
+        service_name: "${{steps.reexport.outputs.service_name}}"
+        otel_resource_attributes: "${{steps.reexport.outputs.otel_resource_attributes}}"
+        shared_actions_repo: "${{steps.reexport.outputs.shared_actions_repo}}"
+        shared_actions_ref: "${{steps.reexport.outputs.shared_actions_ref}}"
+    steps:
+        - id: reexport
+          # NOTE: certs are base-64 encoded so that they're easier to handle here
+          run: |
+            echo endpoint="${OTEL_EXPORTER_OTLP_ENDPOINT}" >> ${GITHUB_OUTPUT}
+            echo service_name="${OTEL_SERVICE_NAME}" >> ${GITHUB_OUTPUT}
+            echo otel_resource_attributes="${OTEL_RESOURCE_ATTRIBUTES}" >> ${GITHUB_OUTPUT}
+            echo shared_actions_ref="${SHARED_ACTIONS_REF}" >> ${GITHUB_OUTPUT}
+            echo shared_actions_repo="${SHARED_ACTIONS_REPO}" >> ${GITHUB_OUTPUT}
+  top-level-telemetry-traceparent:
+    runs-on: ubuntu-latest
+    needs:
+        - reexports
+    outputs:
+      traceparent: "${{steps.traceparent.outputs.traceparent}}"
+    env:
+        OTEL_SERVICE_NAME: ${{needs.reexports.outputs.service_name}}
+    steps:
+        - name: Clone shared-actions repo
+          uses: actions/checkout@v4
+          with:
+            repository: ${{env.SHARED_ACTIONS_REPO}}
+            ref: ${{env.SHARED_ACTIONS_REF}}
+            path: ./shared-actions
+        - id: traceparent
+          uses: ./shared-actions/telemetry-traceparent
```

Comment:
Must we clone this repository? Can we not use a "shared action" as usual?

Something like `uses: rapidsai/shared-actions/...@main`?

Reply:
Yes, we must clone it. There's an explanation for this change at https://github.com/rapidsai/shared-actions. TL;DR - this is the only way to allow shared actions to depend on other shared actions.